### PR TITLE
Monotonic rational quadratic spline

### PR DESCRIPTION
**Pull request by stefanwebb**
Previously, `Spline` and `ConditionalSpline` could only use monotonic rational *linear* splines. I've added the functionality to use quadratic splines, as per the Neural Spline Flow paper

### CODE REVIEW

**Comment by stefanwebb** [comment]
Hmm, there is some problem with Travis and the version of PyTest...

**Comment by fritzo** [comment]
@stefanwebb I've fixed the pytest problems in #2523. You could wait until that merges or pull in the changes (setup.py and tests/integration_tests/test_conjugate_gaussian_models.py).

File: tests/distributions/test_transforms.py [hunk 1]
```diff
@@ -196,7 +196,8 @@ def test_conditional_radial(self):
         self._test_conditional(T.conditional_radial, inverse=False)
 
     def test_conditional_spline(self):
-        self._test_conditional(T.conditional_spline)
+        for order in ['linear', 'quadratic']:
+            self._test_conditional(partial(T.conditional_spline, order=order))
```
Comment (fritzo) [inline tests/distributions/test_transforms.py:200]: nit: Does pytest notation work here?
```py
@pytest.mark.parametrize("order", ["linear", "quadratic"])
def test_conditional_spline(self, order):
    self._test_conditional(partial(T.conditional_spline, order=order))
```
That generally gives clearer test reports.

File: tests/distributions/test_transforms.py [hunk 2]
```diff
@@ -261,7 +262,8 @@ def test_radial(self):
         self._test(T.radial, inverse=False)
 
     def test_spline(self):
-        self._test(T.spline)
+        for order in ['linear', 'quadratic']:
+            self._test(partial(T.spline, order=order))
```
Comment (fritzo) [inline tests/distributions/test_transforms.py:266]: ditto

**Comment by fritzo** [comment]
You should be able to merge with dev now, to fix travis tests.

File: tests/distributions/test_transforms.py (hunk 1 above)
Comment (stefanwebb) [reply tests/distributions/test_transforms.py:200]: It appears you can't do this on class methods: https://stackoverflow.com/questions/18182251/does-pytest-parametrized-test-work-with-unittest-class-based-tests?noredirect=1&lq=1

A workaround would involve requiring an extra dependency... It is better to keep as is?

File: tests/distributions/test_transforms.py (hunk 2 above)
Comment (stefanwebb) [reply tests/distributions/test_transforms.py:266]: ditto

File: tests/distributions/test_transforms.py (hunk 1 above)
Comment (fritzo) [reply tests/distributions/test_transforms.py:200]: fine to keep as is.

**Pull request merged**